code/game/objects/items/explosives/mine.dm

```diff
@@ -1,277 +1,1216 @@
+		result = TRUE
+	return result
+
+//Radiation dust effects
+/particles/radiation
+	icon = 'icons/effects/particles/generic_particles.dmi'
+	icon_state = "cross"
+	width = 100
+	height = 100
+	count = 1000
+	spawning = 0.5
+	lifespan = 30
+	fade = 5
+	fadein = 5
+	position = generator(GEN_CIRCLE, 10, 10, NORMAL_RAND)
+	drift = generator(GEN_VECTOR, list(-0.01, -0.03), list(0.01, 0.03))
+	scale = list(0.5, 0.5)
+	spin = 5
+	color = COLOR_GREEN
+
+/obj/item/mine/shock
+	name = "tesla mine"
+	desc = "Delivers high voltage arcs of lightning at nearby conductive targets. Can be recharged."
+	icon_state = "tesla"
+	range = 3
+	duration = -1
+	disarm_delay = 4 SECONDS
+	undeploy_delay = 5 SECONDS
+	deploy_delay = 5 SECONDS
+	mine_features = MINE_STANDARD_FLAGS|MINE_REUSABLE|MINE_VOLATILE_DAMAGE|MINE_VOLATILE_EMP
+	///The internal cell powering it
+	var/obj/item/cell/battery
+	///How much energy is drained from the internal cell
+	var/energy_cost = 100	//Average cell holds 1000, so 10 shots
+	///How long between each shot
+	var/fire_delay = 0.75 SECONDS
+	///Damage dealt per shot
+	var/damage = 50
+
+/obj/item/mine/shock/Initialize()
+	. = ..()
+	if(battery)
+		battery = new battery(src)
+
+/obj/item/mine/shock/examine(mob/user)
+	. = ..()
+	. += span_notice("[battery ? "Battery Charge - [PERCENT(battery.charge/battery.maxcharge)]%" : "No battery installed."]")
+
+/obj/item/mine/shock/attackby(obj/item/I, mob/user, params)
+	if(!iscell(I))
+		return ..()
+	if(battery)
+		balloon_alert(user, "Already has one!")
+		return
+	user.transferItemToLoc(I, src)
+	battery = I
+	update_icon()
+
+/obj/item/mine/shock/screwdriver_act(mob/living/user, obj/item/I)
+	if(!battery)
+		return balloon_alert(user, "No battery installed!")
+	user.put_in_hands(battery)
+	battery = null
+	update_icon()
+
+/obj/item/mine/shock/trip_mine(atom/movable/victim)
+	if(!battery?.charge)
+		return FALSE
+	return ..()
+
+/obj/item/mine/shock/extra_effects(atom/movable/victim)
+	if(!battery?.charge || battery.charge < energy_cost)
+		playsound(loc, 'sound/machines/buzz-sigh.ogg', 50, sound_range = 7)
+		balloon_alert_to_viewers("Out of charge!")
+		return disarm()
+
+	//Grab a list of nearby objects, shuffle it, then see if they are an eligible victim
+	var/target
+	var/list/nearby_objects = shuffle(circle_range(src, range))
+	nearby_objects -= src	//Prevent the mine from committing suicide
+
+	for(var/atom in nearby_objects)
+		if(!check_path(src, atom))	//circle_range() goes through walls, so make sure the target is reachable
+			continue
+
+		if(isliving(atom))
+			if(ishuman(atom))
+				var/mob/living/carbon/human/human_victim = atom
+				//Will shock a random body part on humans
+				human_victim.apply_damage(damage, BURN, pick(GLOB.human_body_parts), ENERGY)
+				target = human_victim
+			else
+				var/mob/living/living_victim = atom
+				living_victim.apply_damage(damage, BURN, blocked = ENERGY)
+				target = living_victim
+			break
+
+		else if(isobj(atom) && !iseffect(atom))
```

Comment:
Don't think you need the else here since the previous statement breaks if its true